File: TapperObjC/ViewController.m


#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *tapsL;
@property (weak, nonatomic) IBOutlet UIImageView *tapperImgView;
@property (weak, nonatomic) IBOutlet UITextField *textFieldOutlet;
@property (weak, nonatomic) IBOutlet UIButton *playBtn;
@property (weak, nonatomic) IBOutlet UIButton *coinBtn;

@end

@implementation ViewController

- (void)updateTapsLbl {
    NSString *tapsStr = [NSString stringWithFormat:@"%ld", (long)currentTaps];
    NSString *tapStrCombined = [@"Taps: " stringByAppendingString:tapsStr];
    [_tapsL setText:tapStrCombined];
}

- (BOOL)isGameOver {
    return currentTaps >= maxTaps;
}

- (void)restartGame {
    maxTaps = 0;
    _tapsL.text = @"";
    
    [_tapperImgView setHidden:NO];
    [_playBtn setHidden:NO];
    [_textFieldOutlet setHidden:NO];
    
    [_coinBtn setHidden:YES];
    [_tapsL setHidden:YES];
}

- (IBAction)coinPress:(UIButton *)sender {
    [super viewDidLoad];
    currentTaps ++;
    [self updateTapsLbl];
    if ([self isGameOver]) {
        [self restartGame];
    }
}

- (IBAction)playPress:(UIButton *)sender {
    
    if ([_textFieldOutlet text] != NULL && ![_textFieldOutlet.text  isEqual: @""]) {
        [_tapperImgView setHidden:YES];
        [_playBtn setHidden:YES];
        [_textFieldOutlet setHidden:YES];
        
        [_coinBtn setHidden:NO];
        [_tapsL setHidden:NO];
        
        maxTaps =  [_textFieldOutlet.text intValue];
        currentTaps = 0;
        
        [self updateTapsLbl];
        
    }
}

- (IBAction)textField:(UITextField *)sender {
    
}

@end

File: SwiftVersion/tapper-extreme/ViewController.swift

import UIKit

class ViewController: UIViewController {
    
    var maxTaps = 0
    var currentTaps = 0
    
    @IBOutlet weak var logoImg: UIImageView!
    @IBOutlet weak var howManyTapsTxt: UITextField!
    @IBOutlet weak var playBtn: UIButton!
    
    @IBOutlet weak var tapBtn: UIButton!
    @IBOutlet weak var tapsLbl: UILabel!
    
    
    @IBAction func onCoinTapped(_ sender: UIButton) {
        currentTaps += 1
        updateTapsLbl()
        if isGameOver() {restartGame()}
    }

    @IBAction func onPlayBtnPress(_ sender: UIButton) {
        if howManyTapsTxt.text != nil && howManyTapsTxt.text != "" {
            
            logoImg.isHidden = true
            playBtn.isHidden = true
            howManyTapsTxt.isHidden = true
            
            tapBtn.isHidden = false
            tapsLbl.isHidden = false
            
            maxTaps = Int(howManyTapsTxt.text!)!
            currentTaps = 0
            
            updateTapsLbl()
        }
    }

    func restartGame() {
        maxTaps = 0
        howManyTapsTxt.text = ""
        
        logoImg.isHidden = false
        playBtn.isHidden = false
        howManyTapsTxt.isHidden = false
        
        tapBtn.isHidden = true
        tapsLbl.isHidden = true
        
    }
    
    func isGameOver() -> Bool {
        return currentTaps >= maxTaps
    }
    
    func updateTapsLbl() {
        tapsLbl.text = "\(currentTaps) Taps"
    }
}

